### DISCUSSION TITLE
How to update WinGet using the system scope?

### DISCUSSION THREAD

**Discussion by agiacomolli** [microsoft/winget-cli#4512] [Q&A]
I'm using WinGet from a Windows service, and all the WinGet commands run as the system scope, but I am unable to do a self-update:

```
> winget --version
v1.6.3482
> winget update winget --scope machine
No available upgrade found.
No newer package versions are available from the configured sources.
```

On the other hand, it works without the machine scope:

```
> winget update winget
Found App Installer [Microsoft.AppInstaller] Version 1.22.11261.0
...
> winget --version
v1.7.11261
```

Adding more context, the service runs as the SYSTEM user and, at the very beginning, it installs the WinGet dependencies and the latest release from GitHub. I can install and update other packages, but it doesn’t work for WinGet. As a workaround, I need to check if there is a new version on GitHub and reinstall it if it doesn’t match the local one.

Is there a way to use the `winget` command to do a self-update using the machine scope?

**Comment by denelon** [comment]
WinGet is distributed via the "App Installer". It's an MSIX packaged application, so the notion of "per user" or "per machine" is a bit different. The default mechanism for MSIX packages is that they can be registered for any user.

**Comment by yannara** [comment]
Try this:

Invoke-WebRequest 'https://cdn.winget.microsoft.com/cache/source.msix' -OutFile c:\Temp\source.msix
Add-AppxProvisionedPackage -Online -PackagePath 'c:\Temp\source.msix' -SkipLicense
